pkg/term/term_emulator.go

```diff
@@ -0,0 +1,210 @@
+package term
+
+import (
+	"io"
+	"strconv"
+	"strings"
+)
+
+// http://manpages.ubuntu.com/manpages/intrepid/man4/console_codes.4.html
+const (
+	ANSI_ESCAPE_PRIMARY   = 0x1B
+	ANSI_ESCAPE_SECONDARY = 0x5B
+	ANSI_COMMAND_FIRST    = 0x40
+	ANSI_COMMAND_LAST     = 0x7E
+	ANSI_PARAMETER_SEP    = ";"
+	ANSI_CMD_G0           = '('
+	ANSI_CMD_G1           = ')'
+	ANSI_CMD_G2           = '*'
+	ANSI_CMD_G3           = '+'
+	ANSI_CMD_DECPNM       = '>'
+	ANSI_CMD_DECPAM       = '='
+	ANSI_CMD_OSC          = ']'
+	ANSI_CMD_STR_TERM     = '\\'
+	ANSI_BEL              = 0x07
+	KEY_EVENT             = 1
+)
+
+// Terminal contains stdin, stdout and stderr
+type Terminal struct {
+	StdOut io.Writer
+	StdErr io.Writer
+	StdIn  io.ReadCloser
```

Comment:
I agree that it could be a better name. 

VT1002 was an actual terminal and There are many places where VT100  and ANSI are used synonymously. 
In fact  http://en.wikipedia.org/wiki/VT100
"The VT100 is a video terminal, introduced in August 1978 ....
...This led to rapid uptake of the ANSI standard, becoming the de facto standard for terminal emulators."

I think for the programmers coming from c/c++ background (like me), the name stdio has a specific connotation. Here I want to capture the concept of bundling together of stdin+stdout+stderr so used the name Terminal.

How about "Console" ?  

A slight diversion --
(A bit of interesting history .."The Linux console was one of the first features of the kernel and was originally written by Linus Torvalds in 1991[4]"
http://en.wikipedia.org/wiki/Linux_console
http://en.wikipedia.org/wiki/Linux_console#Control_characters